hs_core/HydroRealtimeSignalProcessor.py

```diff
@@ -0,0 +1,110 @@
+from django.db import models
+from haystack.signals import RealtimeSignalProcessor
+from haystack.exceptions import NotHandled
+import logging
+import types
+
+logger = logging.getLogger(__name__)
+logger.debug("load real time processor")
+
+
+class HydroRealtimeSignalProcessor(RealtimeSignalProcessor):
+
+    """ 
+    Customized for the fact that all indexed resources are subclasses of BaseResource. 
+
+    Notes: 
+    1. RealtimeSignalProcessor already plumbs in all class updates. We might want to be more specific. 
+    2. The class sent to this is a subclass of BaseResource, or another class. 
+    3. Thus, we want to capture cases in which it is an appropriate instance, and respond. 
+    """
+
+    def hydro_resource_is_present(self, instance): 
+        """ 
+        This returns True if a current resource is now present in Haystack, and False if not. 
+        This -- in turn -- informs us as to whether to update and/or delete the object 
+        """
+        from haystack.query import SearchQuerySet
+        from haystack.utils import get_identifier
+        discoverable = SearchQuerySet().filter(id=get_identifier(instance)).count()
+        if discoverable > 0:
+            return True
+        else:
+            return False
+
+    def handle_save(self, sender, instance, **kwargs):
+        """
+        Given an individual model instance, determine which backends the
+        update should be sent to & update the object on those backends.
+        """
+        from hs_core.models import BaseResource
+        from hs_access_control.models import ResourceAccess
+        from hs_labels.models import ResourceLabels
```

Comment:
@xuezhaokun We should move these imports back to the top of the file rather than importing them locally in the method, unless there is a circular import issue, in which case, we only locally import the modules which cause circular import issue.